android/java/org/chromium/chrome/browser/crypto_wallet/model/OnboardingViewModel.java

```diff
@@ -5,20 +5,56 @@
 
 package org.chromium.chrome.browser.crypto_wallet.model;
 
+import android.util.Pair;
+import android.util.SparseArray;
+
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.lifecycle.ViewModel;
 
 import org.chromium.brave_wallet.mojom.NetworkInfo;
+import org.chromium.chrome.browser.crypto_wallet.fragments.onboarding.OnboardingVerifyRecoveryPhraseFragment.VerificationStep;
 
+import java.util.ArrayList;
 import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
 import java.util.Set;
 
 public class OnboardingViewModel extends ViewModel {
+    boolean mLegacyRestoreEnabled;
+
     @Nullable private String mPassword;
+    @Nullable private String mRecoveryPhrase;
 
     @NonNull final Set<NetworkInfo> mSelectedNetworks = new HashSet<>();
     @NonNull final Set<NetworkInfo> mAvailableNetworks = new HashSet<>();
+    @NonNull final SparseArray<String> mVerificationWords = new SparseArray<>(3);
+
+    public void setLegacyRestoreEnabled(final boolean legacyRestoreEnabled) {
+        mLegacyRestoreEnabled = legacyRestoreEnabled;
+    }
+
+    public boolean isLegacyRestoreEnabled() {
+        return mLegacyRestoreEnabled;
+    }
+
+    public void setRecoveryPhrase(@NonNull final String recoveryPhrase) {
+        mRecoveryPhrase = recoveryPhrase;
+    }
+
+    @Nullable
+    public String getRecoveryPhrase() {
+        return mRecoveryPhrase;
+    }
+
+    @NonNull
+    public String requireRecoveryPhrase() {
+        if (mRecoveryPhrase == null) {
+            throw new IllegalStateException("Wallet recovery phrase must not be null.");
```

Comment:
do we catch that exception on places where we call `requireRecoveryPhrase`?

Reply:
No we don't. It was an extra check I've introduced just to catch early an error that we may introduce during development/debugging. So better switch it to an assertion.

Reply:
Fixed in commit [`a01af24` (#24943)](https://github.com/brave/brave-core/pull/24943/commits/a01af24cd545c8bf0c04b9e106c45d8f124f9a2c)